packages/contracts/src/dollar/facets/UbiquityPoolFacet.sol

```diff
@@ -149,6 +149,24 @@ contract UbiquityPoolFacet is IUbiquityPool, Modifiers {
         LibUbiquityPool.setCollateralPrice(collateralIndex, newPrice);
     }
 
+    /// @inheritdoc IUbiquityPool
+    function setCollateralChainLinkPriceFeedAddress(
+        address collateralAddress,
+        address chainLinkPriceFeedAddress
+    ) external onlyAdmin {
+        LibUbiquityPool.setCollateralChainLinkPriceFeedAddress(
+            collateralAddress,
+            chainLinkPriceFeedAddress
+        );
+    }
+
+    /// @inheritdoc IUbiquityPool
+    function updateChainLinkCollateralPrice(
```

Comment:
We could remove the `onlyAdmin` modifier because, as far as I understand, there is no issue if somebody else would fetch the latest collateral price.

Another point is that collateral prices should be updated automatically before any collateral price related operations. I couldn't find this feature in the https://github.com/ubiquity/ubiquity-dollar/blob/dac1794b832a8883fa15794eb5afa7f4054680a8/packages/contracts/src/dollar/libraries/LibUbiquityPool.sol

Reply:
Moved `updateChainLinkCollateralPrice` to public functions, now I need to add updates before price related operations.

Reply:
@rndquu I found relevant `getDollarInCollateral` and `collateralUsdBalance` functions that use collateral price. I will update prices automatically in those two. Please let me know if you know other functions where the collateral price should be automatically updated.

Reply:
Those are view methods for UI. I think we should update collateral prices only in `mintDollar()` and `redeemDollar()` methods.

Reply:
Actually, `getDollarInCollateral` and `collateralUsdBalance` are view functions, the price should be updated before the call.

Reply:
> Those are view methods for UI. I think we should update collateral prices only in `mintDollar()` and `redeemDollar()` methods.

All right! I read your comment after pasting mine after I had similar conclusions, will update `mintDollar()` and `redeemDollar()` as well.

Reply:
done!

Reply:
Ooops this had a side effect and messed up related tests, obviously. I need to fix those.

Reply:
All tests fixed with mocked ChainLink price feed.